src/envoy/properties.rs

```diff
@@ -0,0 +1,201 @@
+            Box::new(TypedProperty::string),
+        );
+        properties.insert(
+            "connection.subject_peer_certificate".to_string(),
+            Box::new(TypedProperty::string),
+        );
+        properties.insert(
+            "connection.dns_san_local_certificate".to_string(),
+            Box::new(TypedProperty::string),
+        );
+        properties.insert(
+            "connection.dns_san_peer_certificate".to_string(),
+            Box::new(TypedProperty::string),
+        );
+        properties.insert(
+            "connection.uri_san_local_certificate".to_string(),
+            Box::new(TypedProperty::string),
+        );
+        properties.insert(
+            "connection.uri_san_peer_certificate".to_string(),
+            Box::new(TypedProperty::string),
+        );
+        properties.insert(
+            "connection.sha256_peer_certificate_digest".to_string(),
+            Box::new(TypedProperty::string),
+        );
+        properties.insert(
+            "ratelimit.domain".to_string(),
+            Box::new(TypedProperty::string),
+        );
+
+        properties.insert("request.size".to_string(), Box::new(TypedProperty::integer));
+        properties.insert("source.port".to_string(), Box::new(TypedProperty::integer));
+        properties.insert(
+            "destination.port".to_string(),
+            Box::new(TypedProperty::integer),
+        );
+        properties.insert(
+            "connection.id".to_string(),
+            Box::new(TypedProperty::integer),
+        );
+        properties.insert(
+            "ratelimit.hits_addend".to_string(),
+            Box::new(TypedProperty::integer),
+        );
+
+        properties.insert("metadata".to_string(), Box::new(TypedProperty::metadata));
+
+        properties.insert(
+            "request.headers".to_string(),
+            Box::new(TypedProperty::string_map),
+        );
+        properties.insert(
+            "request.context_extensions".to_string(),
+            Box::new(TypedProperty::string_map),
+        );
+        properties.insert(
+            "source.labels".to_string(),
+            Box::new(TypedProperty::string_map),
+        );
+        properties.insert(
+            "destination.labels".to_string(),
+            Box::new(TypedProperty::string_map),
+        );
+        properties.insert(
+            "filter_state".to_string(),
+            Box::new(TypedProperty::string_map),
+        );
+
+        properties.insert(
+            "auth.metadata".to_string(),
+            Box::new(TypedProperty::complex_map),
+        );
+        properties.insert(
+            "auth.authorization".to_string(),
+            Box::new(TypedProperty::complex_map),
+        );
+        properties.insert(
+            "auth.response".to_string(),
+            Box::new(TypedProperty::complex_map),
+        );
+        properties.insert(
+            "auth.callbacks".to_string(),
+            Box::new(TypedProperty::complex_map),
+        );
```

Comment:
`auth.*` are not properties that can be requested directly to the host through the ABI, right? For that reason they were unchecked for "RL" in [RFC 0002](https://github.com/Kuadrant/architecture/blob/main/rfcs/0002-well-known-attributes.md#auth-attributes) for now. Seeing them here makes me think we may want to progress to the point where these attributes are abstractions that translate to `metadata.filter_metadata.envoy\.filters\.http\.ext_authz.*` perhaps?

Do we need a special `TypedProperty` function to handle them maybe?

Another option would be the "translation" to happen in the policy controller, and therefore the wasm shim won't ever even need to know about them.

Reply:
This actually merely defines the typing... but you are right that I conflated the typing with the host. `Filter::get_typed_property` is actually what decides where the property is to be sourced. 

Maybe I should treat all auth related one differently... tho they'd still use some typed mapper function just as the envoy properties do... 🤔